### DISCUSSION TITLE
Upload personal atlas

### DISCUSSION THREAD

**Discussion by smluiza** [JingyiGF/HERBS#10] [Q&A] [answered]
Hi, congrats on the tool!! 
I have an MRI template and atlas in .mnc format of the rats we currently use in my lab. I was wondering if it is possible somehow to process them to use on HERBS instead of the Waxholm atlas. Thanks!

**Comment by JingyiGF** [comment]
HI, thank you for opening this Q&A, this is a very interesting question for me. The short answer is yes. HERBS was designed for user defined atlas. As you might know from other Issues that I am modifying the code for processing the atlas right now. This also affects the processing steps for user defined atlas. So I need finish this part first.

In general, users need to have 3 files to define an atlas. One is a MRI voxel type file with shape (x, y, z), a file with the same shape labeled with integers to indicate the segmentations of the brain regions, and an excel file or pickle file that contains the information about the labels. It will be great help for me to adjust the code if you can specify what kind of files you have or missing and describe a little more about your data  :-) If you do not want to public the info., you are welcome to email me for a secret discussion. you can find my email address on the front page of my GitHub. 

I will let you know the details about the files needed and update a recipe in Tutorial about how to load and process user defined atlas in 2 weeks. Will come back to you :-)

**Comment by smluiza** [comment]
Hi, thank you so much for your response! I'm using an open-source atlas, here is the link: https://www.zenodo.org/record/3700210#.YzckKnbMKUk

They provide the MRI, a whole brain mask, and a regional mask in .mnc and .nii formats. If you are unable to download it, I'll be more than happy to send you by email. Thank you again!

**Reply by JingyiGF** [reply]
Hi, Thank you very much for the link. I did not open each of them to check, but I think it is ok. Are leftHem and rightHem half of labels? What HERBS requires in the Excel file are label id, label name, parent label id, and label color. I do not see much hierarchy in the .xlsx file, it is fine, but I do need it to be clear for the hierarchies that exist. For example, are all type(Grey, White, Ventricular) parallel and different regions in the 'region' column are parallel?

All info will be transformed to a .pkl file with keys 

label_info.keys()
dict_keys(['index', 'color', 'label', 'abbrev', 'parent', 'level_indicator'])

If you can make this file, things will be much easier and faster. If you have difficulties to do so, I can help you to make such a file based on your explanation for the atlas. But it will take a while, since I am occupied by a revision and have very limit time working on HERBS :-(

**Comment by JingyiGF** [comment, marked as answer]
update the code for user defined atlas and tutorial. Please read the file in Tutorial folder on GitHub.

**Comment by rjp109** [comment]
Hello,

This is a great implementation. I was wondering if it would also be possible to use a far simplified method, consisting of overlaying just one histological image onto one atlas image? Basically, I would like to use the Paxinos and Watson histological atlas (which I only have in PDF format) since this still has far more delineations than the Waxholm rat atlas currently has (and matching histology to a histological atlas makes more sense). I realise HERBS is more sophisticated than this and suited for volume based atlases but I really like your triangulation method! Is this possible?

**Reply by JingyiGF** [reply]
Hi, 

Thanks for asking. Actually users can upload Paxinos :-) it is a hidden feature in HERBS. will write a tutorial during this week and update you. 

cheers,
Jingyi

**Reply by rjp109** [reply]
Ah, amazing, thank you very much!

**Reply by rjp109** [reply]
Hi Jingyi, is there any update on this?

**Reply by JingyiGF** [reply]
Hi, Sorry for the delay and late response. I have been sick -- ok -- sick for a while now, all work is pending/ delayed. Not so easy season when small ones get sick every other week and so do I. I will try my best to catch up soon. Will keep you update. Many thanks for waiting.

**Reply by rjp109** [reply]
Hello, not to worry. I'm sorry to hear that!! I hope you recover soon.

**Reply by JingyiGF** [reply]
Hi @rjp109, Sorry for such a late reply for this topic. I am now updating the code on GitHub and PyPI. And a tutorial at https://github.com/JingyiGF/HERBS/blob/main/Tutorial/Atlas_Related/upload_slice_atlas.md. Let me know if anything is not clear. Or any idea to make it better. I think #32 in discussion is also a very interesting topic related to what you are looking for. 

Thank you for your interest in HERBS in 2022. Wish you a happy holiday season and see you in 2023!

**Reply by rjp109** [reply]
Hi Jingyi, happy New Year, I hope you have recovered and had a restful break.
This is great thank you very much. 
This is probably a fault of my own but when I try to upload a JPEG image with the load slice button, I get the error 'Loading slice atlas is failed. Please check your image or contact maintainers' . To get my slice of interest I just used the snipping tool to create a jpeg from one of the slices in the PDF - do I perhaps need to do a more official conversion of the PDF atlas to jpeg format for this?
I have updated to the latest version of HERBS.

Best wishes,
Ross

**Reply by rjp109** [reply]
Ah, never mind - I instead used print screen and pasted into Powerpoint before saving the image as jpeg, and this seems to work!

**Comment by David-Omer** [comment]
Thanks for the support Jingyi, we really appreciate it! 
I have a question regarding processing an external Atlas. Can you please explain the functionallity of the "factor(voxel)" option in the Atlas Processor gui? 


![image](https://github.com/JingyiGF/HERBS/assets/87187297/32ecbf8c-d92d-47cb-aef8-15925298e1cf)

Thanks, 
Didi
